mongodump/oplog_dump.go

```diff
@@ -44,40 +43,40 @@ func (dump *MongoDump) determineOplogCollectionName() error {
 }
 
 // getOplogCurrentTime returns the most recent oplog entry
-func (dump *MongoDump) getCurrentOplogTime() (bson.MongoTimestamp, error) {
+func (dump *MongoDump) getCurrentOplogTime() (primitive.Timestamp, error) {
 	mostRecentOplogEntry := db.Oplog{}
-	var tempBSON gbson.Raw
+	var tempBSON bson.Raw
 
 	err := dump.SessionProvider.FindOne("local", dump.oplogCollection, 0, nil, &bson.M{"$natural": -1}, &tempBSON, 0)
 	if err != nil {
-		return 0, err
+		return primitive.Timestamp{}, err
 	}
-	err = gbson.Unmarshal(tempBSON, &mostRecentOplogEntry)
+	err = bson.Unmarshal(tempBSON, &mostRecentOplogEntry)
 	if err != nil {
-		return 0, err
+		return primitive.Timestamp{}, err
 	}
-	return bsonutil.ConvertTimestampToMongoTimestamp(mostRecentOplogEntry.Timestamp), nil
+	return mostRecentOplogEntry.Timestamp, nil
 }
 
 // checkOplogTimestampExists checks to make sure the oplog hasn't rolled over
 // since mongodump started. It does this by checking the oldest oplog entry
 // still in the database and making sure it happened at or before the timestamp
 // captured at the start of the dump.
-func (dump *MongoDump) checkOplogTimestampExists(ts bson.MongoTimestamp) (bool, error) {
+func (dump *MongoDump) checkOplogTimestampExists(ts primitive.Timestamp) (bool, error) {
 	oldestOplogEntry := db.Oplog{}
-	var tempBSON gbson.Raw
+	var tempBSON bson.Raw
 
 	err := dump.SessionProvider.FindOne("local", dump.oplogCollection, 0, nil, &bson.M{"$natural": 1}, &tempBSON, 0)
 	if err != nil {
 		return false, fmt.Errorf("unable to read entry from oplog: %v", err)
 	}
-	err = gbson.Unmarshal(tempBSON, &oldestOplogEntry)
+	err = bson.Unmarshal(tempBSON, &oldestOplogEntry)
 	if err != nil {
 		return false, err
 	}
 
 	log.Logvf(log.DebugHigh, "oldest oplog entry has timestamp %v", oldestOplogEntry.Timestamp)
-	if bsonutil.ConvertTimestampToMongoTimestamp(oldestOplogEntry.Timestamp) > ts {
+	if util.CompareTimestamps(oldestOplogEntry.Timestamp, ts) > 0 {
```

Comment:
Maybe make a `util.TimestampGreaterThan(left, right)` function?  It's a little clearer than knowing how "compare" relates to zero.